newrelic/resource_newrelic_service_level.go

```diff
@@ -0,0 +1,302 @@
+				MaxItems:    1,
+				Elem:        eventsQuerySchema(),
+			},
+		},
+	}
+}
+
+func eventsQuerySchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"from": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "",
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+			},
+			"where": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "",
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+			},
+		},
+	}
+}
+
+func objectivesSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "",
+			},
+			"target": {
+				Type:        schema.TypeFloat,
+				Required:    true,
+				Description: "",
+			},
+			"time_window": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "",
+				MinItems:    1,
+				MaxItems:    1,
+				Elem:        timeWindowSchema(),
+			},
+		},
+	}
+}
+
+func timeWindowSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"rolling": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "",
+				MinItems:    1,
+				MaxItems:    1,
+				Elem:        rollingTimeWindowSchema(),
+			},
+		},
+	}
+}
+
+func rollingTimeWindowSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"count": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				Description:  "",
+				ValidateFunc: intInSlice([]int{1, 7, 14, 28, 30}),
+			},
+			"unit": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "",
+				ValidateFunc: validation.StringInSlice([]string{"DAY"}, false),
+			},
+		},
+	}
+}
+
+func resourceNewRelicServiceLevelCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*ProviderConfig).NewClient
+	guid := d.Get("guid").(string)
+	createInput := expandServiceLevelCreateInput(d)
+
+	if createInput.Events.GoodEvents == nil && createInput.Events.BadEvents == nil {
+		return diag.Errorf("err: Defining a new SLI requires a good or bad events query.")
+	}
+	if createInput.Events.GoodEvents != nil && createInput.Events.BadEvents != nil {
+		return diag.Errorf("err: Only a good or bad events query can be defined for an SLI.")
+	}
+
+	log.Printf("[INFO] Creating New Relic One Service Level %s", createInput.Name)
+
+	created, err := client.ServiceLevel.ServiceLevelCreateWithContext(ctx, common.EntityGUID(guid), createInput)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	identifier := serviceLevelIdentifier{
+		AccountID: createInput.Events.AccountID,
+		ID:        created.ID,
+		GUID:      guid,
+	}
+
+	d.SetId(identifier.String())
+	d.Set("sli_id", created.ID)
```

Comment:
This fails linting, and while I dislike the practice, historically we change this to read `_ = d.Set(....)`